R/pkg/R/install.R

```diff
@@ -293,7 +289,7 @@ sparkCachePath <- function() {
         Sys.getenv("XDG_CACHE_HOME", file.path(Sys.getenv("HOME"), ".cache")), "spark")
     }
   } else {
-    stop(sprintf("Unknown OS: %s", .Platform$OS.type))
+    stop(gettextf("Unknown OS: %s", .Platform$OS.type, domain = "R-SparkR"), domain = NA)
```

Comment:
ditto

Reply:
`gettextf` is built with a translation interface and thus preferable to `sprintf`, this is why it's done this way in base R, e.g.

https://github.com/wch/r-source/blob/trunk/src/library/base/R/dataframe.R#L184

```
as.data.frame.default <- function(x, ...)
    stop(gettextf("cannot coerce class %s to a data.frame",
                  sQuote(deparse(class(x))[1L])),
         domain = NA)
```

In this way, when `tools::update_pkg_po('.')` is run, the string literal in `gettextf` is captured as a message for translation; strings for `sprintf` are not so captured.

`SparkR` doesn't have any translations yet, but IMO it doesn't hurt to align to that possibility.

Reply:
See Diagnostic Messages from [Writing R Extensions](https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Diagnostic-messages):

> Try not to split up messages into small pieces. In C error messages use a single format string containing all English words in the messages.

> In R error messages do not construct a message with `paste` (such messages will not be translated) but via multiple arguments to `stop` or `warning`, or via `gettextf`.

Reply:
Keeping the message as "intact" as possible is also for the benefit of translators -- an intact message can be completely adapted to different grammars.

As there is as yet no translation / offer to translate, so this is not a priority for this PR